extra-examples/GoogleTypeDetection/Function.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2021, Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using CloudNative.CloudEvents;
+using Google.Cloud.Functions.Framework;
+using Google.Cloud.Functions.Hosting;
+using Google.Events.Protobuf.Cloud.PubSub.V1;
+using Google.Events.Protobuf.Cloud.Storage.V1;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GoogleTypeDetection
+{
+    public class Startup : FunctionsStartup
+    {
+        public override void ConfigureServices(WebHostBuilderContext context, IServiceCollection services) =>
+            services.AddSingleton<CloudEventFormatter, GoogleEventFormatter>();
```

Comment:
Open question: can we get to a point where this is the pre-configured default in the Functions Framework so developers don't need to provide a `FunctionsStartup` to for this use case.

Reply:
See the readme for more thoughts on this. We could potentially have an `IGoogleCloudEvent` marker interface that's implemented by all the concrete types, and then decorate that with the formatter reference. Customers would then write an `ICloudEventFunction<IGoogleCloudEvent>` implementation.

Having this as a vanilla default is also easy if we're happy to a) make the Functions Framework depend on the Google Events library; b) favour Google event payloads over other payloads. Lots of options to talk about...